tests/test_scripts.py

```diff
@@ -54,6 +55,21 @@ def test_load_into_counting():
     assert os.path.exists(outfile)
 
 
+def test_load_into_counting_nonwritable():
+    script = scriptpath('load-into-counting.py')
+    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
+
+    outfile = utils.get_test_data('test-nonwritable')
+    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
```

Comment:
The issue I had was that I couldn't get git to play nicely with a file that
couldn't be written to, so my solution was to force the file to be
non-writable so we knew it would behave the way we wanted.

On Sat, Apr 4, 2015, 09:23 Michael R. Crusoe <notifications@github.com>
wrote:

> In tests/test_scripts.py
> <https://github.com/ged-lab/khmer/pull/892#discussion_r27768899>:
>
> > @@ -54,6 +55,21 @@ def test_load_into_counting():
> >      assert os.path.exists(outfile)
> >
> >
> > +def test_load_into_counting_nonwritable():
> > +    script = scriptpath('load-into-counting.py')
> > +    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
> > +
> > +    outfile = utils.get_test_data('test-nonwritable')
> > +    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
>
> You might as well create a new file using tempfile.mkdtemp and setting
> the permissions bits on it. No need to include it in the test-data
> directory if it isn't going to have the permissions it advertises. Though
> git does allow you to check files in with 'a-w' perms.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ged-lab/khmer/pull/892/files#r27768899>.
>